DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi/FrontEnd/src/client/app/stats/stats.controller.js

```diff
@@ -0,0 +1,50 @@
+﻿(function() {
+    'use strict';
+
+    angular.module('app.stats')
+        .controller('statsController', statsController);
+    statsController.$inject = ['statsFactory', '$scope', 'logger'];
+
+    function statsController(statsFactory, $scope, logger) {
+        $scope.rep = '';
+        $scope.series = ['Series A'];
+
+        getCommits().then(function () {
+            logger.info('Got commits from GitHub!');
+        });
+
+        getRepos().then(function() {
+            logger.info('Got repositories');
+        });
+
+        function getCommits() {
+            return statsFactory.getCommitsFromRepos().then(function (response) {
+                $scope.reposData = [response.data.commits];
+                $scope.labels = response.data.months;
+                $scope.barData = [response.data.commits];
+                $scope.eachDate = response.data.commitsForEverRepository;
+                $scope.eachSeries = response.data.repositories.map(function(el) {
+                    return el.name;
+                });
+                $scope.userInfo = response.data.userInfo;
+            });
+        }
+
+        function getRepos() {
+            return statsFactory.getRepos().then(function(response) {
+                $scope.Repos = response.data;
+            });
+        }
+
+        $scope.getCommitsFromCurrent = function (repo) {
+            return statsFactory.getCommitsFromCurrentRepo(repo).then(function (response) {
+                $scope.reposData = [response.data];
```

Comment:
воу-воу-воу. why do you wrap data to array?
it would be better if service do the action it is named for. If you call it `getCommitsFromCurrentRepo` - just return array of commits. not response object. in this case such level of abstraction is useless